src/Core/Output/TypedDataDumper.cs

```diff
@@ -182,7 +182,7 @@ public void VisitString(StringType str)
                     while (rdr.TryReadByte(out byte b))
                     {
                         //$REVIEW: assumes ASCII.
-                        if (0x20 <= b && b < 0x7F)
+                        if (0x20 >= b && b < 0x7F)
```

Comment:
You've introduced a bug. This expression will print out ASCII control characters (those whose code points are < 0x20) which is not the intent of this code. Running the unit tests would have caught this mistake. Please revert this change.